ecosystem/indexer/src/indexer/tailer.rs

```diff
@@ -80,6 +86,48 @@ impl Tailer {
         info!("Migrations complete!");
     }
 
+    /// If chain id doesn't exist, save it. Otherwise make sure that we're indexing the same chain
+    pub async fn check_or_update_chain_id(&self) -> anyhow::Result<bool> {
+        info!("Checking if chain id is correct");
+        let conn = self
+            .connection_pool
+            .get()
+            .expect("DB connection should be available at this stage");
+
+        let query_chain = dsl::ledger_infos
+            .select(dsl::chain_id)
+            .load::<i64>(&conn)
+            .expect("Error loading chain id from db");
+        let maybe_existing_chain_id = query_chain.first();
+
+        let new_chain_id = self
+            .transaction_fetcher
+            .lock()
+            .await
+            .fetch_ledger_info()
+            .await
+            .chain_id as i64;
+
+        match maybe_existing_chain_id {
+            Some(chain_id) => {
+                ensure!(*chain_id == new_chain_id, "Wrong chain detected! Trying to index chain {} now but existing data is for chain {}", new_chain_id, chain_id);
+                info!("Chain id matches! Continuing to index chain {}", chain_id);
+                Ok(true)
+            }
+            None => {
+                info!("Adding chain id {} to db, continue indexing", new_chain_id);
+                execute_with_better_error(
+                    &conn,
+                    diesel::insert_into(ledger_infos::table).values(LedgerInfo {
+                        chain_id: new_chain_id,
+                    }),
+                )
+                .expect("Error updating chain_id!");
```

Comment:
This function returns a result already, use `.context("msg")?` instead.

Reply:
Got error
```
no method named `context` found for enum `Result` in the current scope
```